src/SFA.DAS.AODP.Application/Commands/Qualifications/CreateQualificationDiscussionHistoryCommandHandler.cs

```
@@ -0,0 +1,74 @@
+﻿using MediatR;
+using SFA.DAS.AODP.Data.Entities.Qualification;
+using SFA.DAS.AODP.Data.Repositories.Qualification;
+
+namespace SFA.DAS.AODP.Application.Commands.Qualifications
+{
+    public class CreateQualificationDiscussionHistoryCommandHandler : IRequestHandler<CreateQualificationDiscussionHistoryCommand, BaseMediatrResponse<EmptyResponse>>
+    {
+        private readonly IQualificationDiscussionHistoryRepository _repository;
+        private readonly IQualificationFundingsRepository _qualificationFundingsRepository;
+        private readonly IQualificationRepository _qualificationRepository;
+
+        public CreateQualificationDiscussionHistoryCommandHandler(IQualificationDiscussionHistoryRepository repository, IQualificationFundingsRepository qualificationFundingsRepository, IQualificationRepository qualificationRepository)
+        {
+            _repository = repository;
+            _qualificationFundingsRepository = qualificationFundingsRepository;
+            _qualificationRepository = qualificationRepository;
+        }
+
+        public async Task<BaseMediatrResponse<EmptyResponse>> Handle(CreateQualificationDiscussionHistoryCommand request, CancellationToken cancellationToken)
+        {
+            var response = new BaseMediatrResponse<EmptyResponse>();
+
+            try
+            {
+                if (string.IsNullOrEmpty(request.QualificationReference))
+                {
+                    response.ErrorMessage = "Qualification reference is required";
+                    response.Success = false;
+                    return response;
+                }
+                var qualification = await _qualificationRepository.GetByIdAsync(request.QualificationReference);
+                if (qualification == null)
```

Better to throw an exception here to prevent duplication